.github/workflows/api-compatibility.yml

```diff
@@ -31,9 +31,9 @@ jobs:
           path: ${{ github.head_ref }}
 
       - name: Setup Go
-        uses: actions/setup-go@f111f3307d8850f501ac008e886eec1fd1932a34 # v5.3.0
+        uses: actions/setup-go@0aaccfd150d50ccaeb58ebd88d36e91967a5f35b # v5.4.0
         with:
-          go-version: ~1.23.7
+          go-version: ~1.24.0
```

Comment:
cannot use go1.24 yet

Reply:
Why not for some actions like api compatibility?

Reply:
I asked the same question and got the no answer in https://github.com/open-telemetry/opentelemetry-collector/pull/12655 & https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38726